### DISCUSSION TITLE
How do I know what each circle operation does?

### DISCUSSION THREAD

**Discussion by miusic** [Samsung/ONE#12394] [Q&A] [answered]
Is there any reference to read for the actions that the `circle operations` does?

For instance, if I want to know what the `SparseToDense` operation does and how this operation works, which document should I read to?

If there is helpful reference, please let me know. I truely appriciate it! 🙂

**Comment by jyoungyun** [comment, marked as answer]
AFAIK, almost circle operations came from the TensorFlow operation.
Therefore, you can get a hint about what the circle operation does from TensorFlow guideline.
Since several frameworks provide similar operations or functions, you can refer other framework guidelines.

**Reply by miusic** [reply]
Thank you for the answer! 😄 I'll take a look for the TensorFlow guideline!

**Comment by ragmani** [comment]
This is the guide for operation compatibility of TensorFlow Lite and TensorFlow.
https://www.tensorflow.org/lite/guide/ops_compatibility

And many(most) circle operations are from tflite. If you want to know how each operation works, I recommend the guide https://www.tensorflow.org/mlir/tfl_ops . 
Although, the guide lacks sufficient explanation, it is the closest explanation to circle operations.

**Reply by miusic** [reply]
Thank you for kind reponse! 🙂
